io/src/test/java/io/smallrye/common/io/Files2Test.java

```diff
@@ -0,0 +1,203 @@
+        Files2.cleanRecursively(testArea);
+        assertTrue(Files.exists(testArea));
+        assertFalse(Files.exists(testArea.resolve("blah.bin")));
+        assertFalse(Files.exists(testArea.resolve("empty.txt")));
+        assertTrue(Files.exists(testArea.resolve("subDir")));
+        assertFalse(Files.exists(testArea.resolve("subDir").resolve("subfile")));
+        Files2.deleteRecursively(testArea);
+        assertFalse(Files.exists(testArea));
+    }
+
+    @Test
+    public void testCleanRecursivelyEvenIfInsecure() throws IOException {
+        // first we have to create a bunch of files
+        Path testArea = Path.of("target/test-area");
+        makeStructure(testArea);
+        // rough check to make sure things got created
+        assertTrue(Files.exists(testArea));
+        assertTrue(Files.exists(testArea.resolve("subDir").resolve("subDir2")));
+        Files2.cleanRecursivelyEvenIfInsecure(testArea);
+        assertTrue(Files.exists(testArea));
+        assertFalse(Files.exists(testArea.resolve("blah.bin")));
+        assertFalse(Files.exists(testArea.resolve("empty.txt")));
+        assertTrue(Files.exists(testArea.resolve("subDir")));
+        assertFalse(Files.exists(testArea.resolve("subDir").resolve("subfile")));
+        Files2.deleteRecursivelyEvenIfInsecure(testArea);
+        assertFalse(Files.exists(testArea));
+    }
+
+    @Test
+    public void testDeleteRecursivelyQuiet() {
+        assumeTrue(Files2.hasSecureDirectories());
+        Path testArea = Path.of("target/test-area");
+        makeStructure(testArea);
+        // rough check to make sure things got created
+        assertTrue(Files.exists(testArea));
+        assertTrue(Files.exists(testArea.resolve("subDir").resolve("subDir2")));
+        // do it
+        DeleteStats stats = Files2.deleteRecursivelyQuiet(testArea);
+        assertFalse(Files.exists(testArea));
+        assertEquals(6, stats.filesFound());
+        assertEquals(6, stats.filesRemoved());
+        assertEquals(4, stats.directoriesFound());
+        assertEquals(4, stats.directoriesRemoved());
+    }
+
+    @Test
+    public void testDeleteRecursivelyQuietAbsolute() {
+        assumeTrue(Files2.hasSecureDirectories());
+        Path testArea = Path.of("target/test-area").toAbsolutePath();
+        makeStructure(testArea);
+        // rough check to make sure things got created
+        assertTrue(Files.exists(testArea));
+        assertTrue(Files.exists(testArea.resolve("subDir").resolve("subDir2")));
+        // do it
+        DeleteStats stats = Files2.deleteRecursivelyQuiet(testArea);
+        assertFalse(Files.exists(testArea));
+        assertEquals(6, stats.filesFound());
+        assertEquals(6, stats.filesRemoved());
+        assertEquals(4, stats.directoriesFound());
+        assertEquals(4, stats.directoriesRemoved());
+    }
+
+    @Test
+    public void testDeleteRecursivelyQuietEvenIfInsecure() {
```

Comment:
You may already know, but you can also ask JUnit to create a temporary dir for your tests:


```suggestion
    public void testDeleteRecursivelyQuietEvenIfInsecure(@TempDir Path tempDir) {
```

Reply:
Nice, I'll try that.

Reply:
Ah, unfortunately that always gives an absolute path, which makes it hard to test relative path stuff. But I can still use it for absolute path tests.